lib/iris/analysis/__init__.py

```diff
@@ -1472,6 +1473,64 @@ def _proportion(array, function, axis, **kwargs):
     return result
 
 
+def _lazy_max_run(array, axis=-1, **kwargs):
```

Comment:
I think the dask cumulative reductions are actually able to handle data being chunked along the axis being acted on. In fact I think all operations in `_lazy_max_run_calc` ought not to require `map_complete_blocks`. It's still probably worth doing the `da.squeeze` operation here mind you.

Reply:
Removed `map_complete_blocks`, kept the squeeze